share_app/src/main/java/org/odk/share/controller/WifiHotspot.java

```diff
@@ -0,0 +1,151 @@
+package org.odk.share.controller;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.wifi.WifiConfiguration;
+import android.net.wifi.WifiManager;
+import android.util.Log;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+/**
+ * Created by laksh on 5/17/2018.
+ */
+
+public class WifiHotspot {
+
+    private Method getWifiApConfig;
+    private Method setWifiApEnable;
+    private Method setWifiApConfig;
+    private Method isWifiApEnabled;
+    private Method getWifiApState;
+    private WifiManager wifiManager;
+    private WifiConfiguration lastConfig;
+    private WifiConfiguration currConfig;
+    private static final String ssid = "ODK-Share";
```

Comment:
For now, I've created new wifi configuration with `ssid name= "ODK-Share"` and the hotspot is open(This will be the default config). But when I add the settings there I'll add the option so that user can name the new config and choose the password for the created hotspot.

Reply:
This is looking great.
Next step would be to pick these configurations from shared preferences and creating a simple settings activity to change these settings.